specification/dartLangSpec.tex

```diff
@@ -8320,54 +8719,63 @@ \subsection{Strings}
 \LMHash{}%
 A single line string is delimited by either matching single quotes or matching double quotes.
 
-\commentary{
-Hence, `abc' and ``abc'' are both legal strings, as are `He said ``To be or not to be'' did he not?' and ``He said `To be or not to be' didn't he''.
-However ``This ` is not a valid string, nor is `this''.
+\commentary{%
+Hence, `abc' and ``abc'' are both legal strings,
+as are `He said ``To be or not to be'' did he not?' and
+``He said `To be or not to be' didn't he''.
+However ``This ` is not a valid string, nor is `this''.%
 }
 
-\commentary{
-The grammar ensures that a single line string cannot span more than one line of source code, unless it includes an interpolated expression that spans multiple lines.
+\commentary{%
+The grammar ensures that a single line string cannot span more than
+one line of source code,
+unless it includes an interpolated expression that spans multiple lines.%
 }
 
 \LMHash{}%
 Adjacent strings are implicitly concatenated to form a single string literal.
 
-\commentary{
-Here is an example
+\commentary{%
+Here is an example:%
 }
 
 \begin{dartCode}
 print("A string" "and then another"); // \comment{A stringand then another}
 \end{dartCode}
 
-\rationale{
+\rationale{%
 Dart also supports the operator + for string concatenation.
 
 The + operator on Strings requires a String argument.
 It does not coerce its argument into a string.
-This helps avoid puzzlers such as
+This helps avoid puzzlers such as%
 }
 
 \begin{dartCode}
 print("A simple sum: 2 + 2 = " +
             2 + 2);
 \end{dartCode}
 
-\rationale{
-which this prints 'A simple sum: 2 + 2 = 22' rather than 'A simple sum: 2 + 2 = 4'.
-However, the use of the concatenation operation is still discouraged for efficiency reasons.
-Instead, the recommended Dart idiom is to use string interpolation.
+\rationale{%
+which prints 'A simple sum: 2 + 2 = 22' rather than
+'A simple sum: 2 + 2 = 4'.
+However, the use of the concatenation operation is still discouraged
```

Comment:
That paragraph should be rewritten or removed.
It should be used judiciously, but it is not *discouraged* when used correctly.

Reply:
Adjusted the text to say that string interpolation 'recommended', which leaves concatenation as "unspecified, but not exactly recommended".